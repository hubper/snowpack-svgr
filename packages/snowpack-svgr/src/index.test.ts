import { join } from 'path';
import snowpackSvgr from './index';
import { PluginLoadOptions } from 'snowpack';

describe('snowpack-svgr', () => {
  const snowpackConfig: any = {
    mount: {
      [__dirname]: { url: '/dist' },
    },
    buildOptions: {
      out: 'build',
    },
  };

  it('should compile .svg files', async () => {
    const filePath = join(__dirname, '__fixtures__/test.svg');

    const plugin = snowpackSvgr(snowpackConfig);
    const result = await plugin.load({
      filePath,
      fileExt: '.svg',
    } as PluginLoadOptions);

    expect(result).toMatchSnapshot();
  });

  it('should be to export the url as default export', async () => {
    const filePath = join(__dirname, '__fixtures__/test.svg');

    const plugin = snowpackSvgr(snowpackConfig, { exportUrlAsDefault: true });
    const result = await plugin.load({
      filePath,
      fileExt: '.svg',
    } as PluginLoadOptions);

    expect(result['.svg']).toBeDefined();
    expect(result['.js']).toMatchSnapshot();
  });
});
