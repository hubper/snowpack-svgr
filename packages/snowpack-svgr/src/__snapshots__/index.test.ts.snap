// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snowpack-svgr should be to export the url as default export 1`] = `
"var _path, _rect, _rect2, _rect3;

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

import React from \\"react\\";

function SvgTest(props, svgRef) {
  return /*#__PURE__*/React.createElement(\\"svg\\", _extends({
    width: 187,
    height: 86,
    fill: \\"none\\",
    xmlns: \\"http://www.w3.org/2000/svg\\",
    ref: svgRef
  }, props), _path || (_path = /*#__PURE__*/React.createElement(\\"path\\", {
    d: \\"M72.72 64c12.808 0 22.651-8.538 24.549-20.753 1.778-11.859-5.455-20.753-17.966-20.753h-13.52L58.728 64h13.994zm5.515-33.798c7.471 0 11.622 5.1 10.377 13.045-1.364 8.242-7.65 12.986-15.298 12.986H68.63l4.506-26.03h5.1zm20.831 20.16c-1.008 7.768 3.202 14.172 10.614 14.172 3.972 0 7.352-1.898 9.724-4.922L118.692 64h7.471l4.685-27.275h-7.531l-.889 5.218c-1.364-3.44-4.091-5.752-8.42-5.811-7.352-.06-13.875 6.107-14.942 14.23zm7.886 0c.712-4.328 4.447-7.945 8.657-7.827 3.736.12 5.811 3.736 5.277 7.827-.711 4.329-4.565 7.827-8.36 7.768-4.151-.06-6.167-3.676-5.574-7.768zm28.816 4.27l-4.21 4.684c2.906 3.024 7.175 5.633 12.749 5.633 5.988 0 10.139-3.499 10.435-8.835.237-4.744-2.964-6.7-6.937-8.36-2.728-1.127-5.396-1.839-5.218-3.914.059-1.305 1.364-2.253 3.261-2.253 2.491 0 4.981 1.66 6.819 3.676l3.617-4.684c-2.609-2.49-6.167-4.447-10.614-4.447-5.099 0-9.783 2.609-10.08 7.827-.237 4.624 3.143 6.818 6.582 8.123 2.787 1.008 5.159 2.075 5.099 4.21-.059 1.541-1.482 2.787-3.38 2.787-2.846.118-5.455-1.72-8.123-4.447zm41.314-7.234L174.176 64h7.59l3.024-17.67c.83-5.988-1.186-10.495-7.59-10.495-3.735 0-6.581 1.898-8.479 4.566l3.914-22.65h-7.649L157.04 64h7.59l2.905-16.958c.831-2.787 3.143-5.277 6.286-5.159 3.142.119 3.617 2.61 3.261 5.515z\\",
    fill: \\"currentColor\\"
  })), _rect || (_rect = /*#__PURE__*/React.createElement(\\"rect\\", {
    x: 14,
    y: 40,
    width: 36,
    height: 6,
    rx: 3,
    fill: \\"currentColor\\"
  })), _rect2 || (_rect2 = /*#__PURE__*/React.createElement(\\"rect\\", {
    y: 53,
    width: 36,
    height: 6,
    rx: 3,
    fill: \\"currentColor\\"
  })), _rect3 || (_rect3 = /*#__PURE__*/React.createElement(\\"rect\\", {
    x: 6,
    y: 27,
    width: 36,
    height: 6,
    rx: 3,
    fill: \\"currentColor\\"
  })));
}

var ForwardRef = /*#__PURE__*/React.forwardRef(SvgTest);
export default \\"/dist/__fixtures__/test.svg.svg\\"; export { ForwardRef as ReactComponent };"
`;

exports[`snowpack-svgr should compile .svg files 1`] = `
Object {
  ".js": "var _path, _rect, _rect2, _rect3;

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

import React from \\"react\\";

function SvgTest(props, svgRef) {
  return /*#__PURE__*/React.createElement(\\"svg\\", _extends({
    width: 187,
    height: 86,
    fill: \\"none\\",
    xmlns: \\"http://www.w3.org/2000/svg\\",
    ref: svgRef
  }, props), _path || (_path = /*#__PURE__*/React.createElement(\\"path\\", {
    d: \\"M72.72 64c12.808 0 22.651-8.538 24.549-20.753 1.778-11.859-5.455-20.753-17.966-20.753h-13.52L58.728 64h13.994zm5.515-33.798c7.471 0 11.622 5.1 10.377 13.045-1.364 8.242-7.65 12.986-15.298 12.986H68.63l4.506-26.03h5.1zm20.831 20.16c-1.008 7.768 3.202 14.172 10.614 14.172 3.972 0 7.352-1.898 9.724-4.922L118.692 64h7.471l4.685-27.275h-7.531l-.889 5.218c-1.364-3.44-4.091-5.752-8.42-5.811-7.352-.06-13.875 6.107-14.942 14.23zm7.886 0c.712-4.328 4.447-7.945 8.657-7.827 3.736.12 5.811 3.736 5.277 7.827-.711 4.329-4.565 7.827-8.36 7.768-4.151-.06-6.167-3.676-5.574-7.768zm28.816 4.27l-4.21 4.684c2.906 3.024 7.175 5.633 12.749 5.633 5.988 0 10.139-3.499 10.435-8.835.237-4.744-2.964-6.7-6.937-8.36-2.728-1.127-5.396-1.839-5.218-3.914.059-1.305 1.364-2.253 3.261-2.253 2.491 0 4.981 1.66 6.819 3.676l3.617-4.684c-2.609-2.49-6.167-4.447-10.614-4.447-5.099 0-9.783 2.609-10.08 7.827-.237 4.624 3.143 6.818 6.582 8.123 2.787 1.008 5.159 2.075 5.099 4.21-.059 1.541-1.482 2.787-3.38 2.787-2.846.118-5.455-1.72-8.123-4.447zm41.314-7.234L174.176 64h7.59l3.024-17.67c.83-5.988-1.186-10.495-7.59-10.495-3.735 0-6.581 1.898-8.479 4.566l3.914-22.65h-7.649L157.04 64h7.59l2.905-16.958c.831-2.787 3.143-5.277 6.286-5.159 3.142.119 3.617 2.61 3.261 5.515z\\",
    fill: \\"currentColor\\"
  })), _rect || (_rect = /*#__PURE__*/React.createElement(\\"rect\\", {
    x: 14,
    y: 40,
    width: 36,
    height: 6,
    rx: 3,
    fill: \\"currentColor\\"
  })), _rect2 || (_rect2 = /*#__PURE__*/React.createElement(\\"rect\\", {
    y: 53,
    width: 36,
    height: 6,
    rx: 3,
    fill: \\"currentColor\\"
  })), _rect3 || (_rect3 = /*#__PURE__*/React.createElement(\\"rect\\", {
    x: 6,
    y: 27,
    width: 36,
    height: 6,
    rx: 3,
    fill: \\"currentColor\\"
  })));
}

var ForwardRef = /*#__PURE__*/React.forwardRef(SvgTest);
export default ForwardRef;",
}
`;
